Bluepill-runner/Bluepill-runner/BPTraceEvent.m

```diff
@@ -0,0 +1,60 @@
+//
+//  BPTraceEvent.m
+//  bluepill
+//
+//  Created by Eric Snyder on 12/17/18.
+//  Copyright © 2018 LinkedIn. All rights reserved.
+//
+
+#import "BPTraceEvent.h"
+
+@implementation TraceEvent
+- (instancetype)init {
+    self = [self initWithData:nil];
+    return self;
+}
+
+- (instancetype)initWithData:(NSDictionary *)data {
+    self = [super init];
+    if (self) {
+        _displayTimeUnit = @"ns";
+        _systemTraceEvents = @"SystemTraceData";
+        _otherData = data;
+        _stackFrames = [[NSDictionary alloc] init];
+        _samples = [NSMutableArray array];
+        _traceEvents = [NSMutableArray array];
+    }
+    return self;
+}
+
+- (void)appendCompleteTraceEvent:(NSString *)name
+                                :(NSString *)category
+                                :(NSInteger)timestamp
+                                :(float)duration
+                                :(NSInteger)process_id
+                                :(NSInteger)thread_id
+                                :(NSDictionary *)args {
+    NSDictionary *newTraceEvent = [[NSDictionary alloc] initWithObjectsAndKeys:
+                                   name, @"name",
+                                   category, @"cat",
+                                   @"X", @"ph", // Complete event type (with both a timestamp and duration)
+                                   [NSString stringWithFormat: @"%ld", (long)timestamp], @"ts",
+                                   [[NSNumber numberWithFloat:duration] stringValue], @"dur",
```

Comment:
"dur" -> "duration"?

Reply:
See above.

Reply:
Same, part of the [spec](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview)